components/supervisor/pkg/supervisor/services.go

```diff
@@ -701,6 +706,53 @@ func (c *ControlService) ExposePort(ctx context.Context, req *api.ExposePortRequ
 	return &api.ExposePortResponse{}, err
 }
 
+// CreateSSHKeyPair create a ssh key pair for the workspace.
+func (ss *ControlService) CreateSSHKeyPair(context.Context, *api.CreateSSHKeyPairRequest) (response *api.CreateSSHKeyPairResponse, err error) {
+	if ss.privateKey != "" && ss.publicKey != "" {
+		return &api.CreateSSHKeyPairResponse{
+			PrivateKey: ss.privateKey,
+		}, nil
+	}
+	home, _ := os.UserHomeDir()
+	dir, err := os.MkdirTemp(os.TempDir(), "ssh-key-*")
+	if err != nil {
+		return nil, xerrors.Errorf("cannot create tmpfile: %w", err)
+	}
+	err = prepareSSHKey(context.Background(), filepath.Join(dir, "ssh"))
+	if err != nil {
+		return nil, xerrors.Errorf("cannot create ssh key pair: %w", err)
+	}
+	bPublic, err := os.ReadFile(filepath.Join(dir, "ssh.pub"))
+	if err != nil {
+		return nil, xerrors.Errorf("cannot read publickey: %w", err)
+	}
+	bPrivate, err := os.ReadFile(filepath.Join(dir, "ssh"))
+	if err != nil {
+		return nil, xerrors.Errorf("cannot read privatekey: %w", err)
+	}
+	err = os.MkdirAll(filepath.Join(home, ".ssh"), 0o751)
```

Comment:
Just wondering about the permissions. `0751` is a bit unusual. The [man page](https://linux.die.net/man/1/ssh) recommends `0700`. What's the motivation for this choice?